scipy/optimize/tests/test_bracket.py

```diff
@@ -286,15 +286,387 @@ def f(x):
 
         # 3. bracket limit hits root exactly
         with np.errstate(over='ignore'):
-            res = _bracket_root(f, 5, 10, min=0)
+            res = _bracket_root(f, 5, 10, xmin=0)
         bracket = (res.xl, res.xr)
         assert_allclose(bracket[0], 0, atol=1e-15)
         with np.errstate(over='ignore'):
-            res = _bracket_root(f, -10, -5, max=0)
+            res = _bracket_root(f, -10, -5, xmax=0)
         bracket = (res.xl, res.xr)
         assert_allclose(bracket[1], 0, atol=1e-15)
 
         # 4. bracket not within min, max
         with np.errstate(over='ignore'):
-            res = _bracket_root(f, 5, 10, min=1)
+            res = _bracket_root(f, 5, 10, xmin=1)
         assert not res.success
+
+
+class TestBracketMinimum:
+    def init_f(self):
+        def f(x, xl0, xr0):
+            f.count += 1
+            return (x - xl0)**2 + xr0
+        f.count = 0
+        return f
+
+    def assert_valid_bracket(self, result):
+        assert np.all(
+            (result.xl < result.xm) & (result.xm < result.xr)
+        )
+        assert np.all(
+            (result.fl >= result.fm) & (result.fr > result.fm)
+            | (result.fl > result.fm) & (result.fr > result.fm)
+        )
+
+    def get_kwargs(
+            self, *, xl0=None, xr0=None, factor=None, xmin=None, xmax=None, args=()
+    ):
+        names = ("xl0", "xr0", "xmin", "xmax", "factor", "args")
+        return {
+            name: val for name, val in zip(names, (xl0, xr0, xmin, xmax, factor, args))
+            if isinstance(val, np.ndarray) or np.isscalar(val)
+            or val not in [None, ()]
+        }
+
+    @pytest.mark.parametrize(
+        "seed",
+        (
+            307448016549685229886351382450158984917,
+            11650702770735516532954347931959000479,
+            113767103358505514764278732330028568336,
+        )
+    )
+    @pytest.mark.parametrize("use_xmin", (False, True))
+    @pytest.mark.parametrize("other_side", (False, True))
+    def test_nfev_expected(self, seed, use_xmin, other_side):
```

Comment:
LMK if you want me to check the details, but like `TestBracketRoot.test_nfev_expected`, this looks like a very strong test that confirms the algorithm is behaving precisely as expected in the scalar case.